### PR TITLE
merge bitcoin#16787, #17474, #17687, #18165, #18877, #18960, #19010, #19044, #19070: block filters

### PR DESCRIPTION
**Pull request by kittywhiskers**

### CODE REVIEW

**Comment by PastaPastaPasta** [comment]
linter + both tests are unhappy

**Comment by kittywhiskers** [comment]
It's a work in progress

**Comment by PastaPastaPasta** [comment]
Marked as draft

**Comment by PastaPastaPasta** [comment]
Please rename all commits to begin with "Merge"

File: src/net_processing.cpp [hunk 1]
```diff
@@ -28,6 +30,7 @@
 #include <index/txindex.h>
 #include <txmempool.h>
 #include <ui_interface.h>
+#include <validation.h>
```
Comment (PastaPastaPasta) [inline src/net_processing.cpp:33]: This should be below L16 to be inline with upstream

File: src/net_processing.h [hunk 2]
```diff
@@ -21,6 +21,7 @@ static const unsigned int DEFAULT_BLOCK_RECONSTRUCTION_EXTRA_TXN = 100;
 /** Default for BIP61 (sending reject messages) */
 static constexpr bool DEFAULT_ENABLE_BIP61 = true;
 static const bool DEFAULT_PEERBLOOMFILTERS = true;
+static const bool DEFAULT_PEERBLOCKFILTERS = false;
```
Comment (PastaPastaPasta) [inline src/net_processing.h:24]: We should consider what all of these defaults should be, like should normal nodes be serving bloom filters by default for example?

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]
LGTM, besides one thing and a comment

File: src/init.cpp [hunk 3]
```diff
@@ -555,6 +555,7 @@ void SetupServerArgs()
     gArgs.AddArg("-onlynet=<net>", "Make outgoing connections only through network <net> (ipv4, ipv6 or onion). Incoming connections are not affected by this option. This option can be specified multiple times to allow multiple networks.", false, OptionsCategory::CONNECTION);
     gArgs.AddArg("-peerbloomfilters", strprintf("Support filtering of blocks and transaction with bloom filters (default: %u)", DEFAULT_PEERBLOOMFILTERS), false, OptionsCategory::CONNECTION);
     gArgs.AddArg("-peertimeout=<n>", strprintf("Specify p2p connection timeout in seconds. This option determines the amount of time a peer may be inactive before the connection to it is dropped. (minimum: 1, default: %d)", DEFAULT_PEER_CONNECT_TIMEOUT), false, OptionsCategory::CONNECTION);
+    gArgs.AddArg("-peerblockfilters", strprintf("Serve compact block filters to peers per BIP 157 (default: %u)", DEFAULT_PEERBLOCKFILTERS), false, OptionsCategory::CONNECTION);
```
Comment (UdjinM6) [inline src/init.cpp:558]: Should be one above `-peerbloomfilters` (sort A-Z)

File: test/functional/test_framework/mininode.py [hunk 4]
```diff
@@ -31,6 +31,7 @@
     msg_block,
     msg_blocktxn,
     msg_clsig,
+    msg_cfcheckpt,
```
Comment (UdjinM6) [inline test/functional/test_framework/mininode.py:34]: Should be above `msg_clsig` (A-Z)

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
Few small issues

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**